pkg/adapter/v2/cloudevents.go

```diff
@@ -113,6 +120,23 @@ type client struct {
 
 var _ cloudevents.Client = (*client)(nil)
 
+// SendCloudEvent prepares logging while sending an event for APIServerSource and PingSource
+// and sends the event to client.Send
+func SendCloudEvent(ctx context.Context, event cloudevents.Event, logger *zap.SugaredLogger, ce cloudevents.Client) {
+	event.SetID(uuid.New().String()) // provide an ID here so we can track it with logging
+	defer logger.Debug("Finished sending cloudevent id: ", event.ID())
+	if event.Context.GetType() == "dev.knative.sources.ping" {
```

Comment:
The intent is to make SendCloudEvent completely generic, so this does not sound right.

Since factorizing out the common code is not so trivial, in particular due to the fact that the APIServerSource is single-tenant and the PingSource is multi-tenant, maybe the best is to not do this in this PR. WDYT?

Reply:
I'd agree that the SendCloudEvent is not completely generic right now, and the different models for both of these sources has a part to play, ideally the context should carry the target so as to not do if-else or switching but also, it can be converted into completely generic once the APIServerSource is changed to multi-tenant later on (if thats in cards?) - which would be a simple change I believe. IMO this factorizing is still an improvement and atleast sets the direction right. 

But your call really, I am not aware of the bigger picture. I can revert back to https://github.com/knative/eventing/pull/4723/commits/d2440ed407bf7e1709dfd442bf02e9ab6a1ecf18, + printing target here (using the current way of making use of cloudeventTarget variable to store the target) + one logline "cloudevent sent"

Reply:
Hmm, if I go back to the previous commit, I'd remove the target completely, using a global variable could be problematic. 
I see your point more now, the refactoring is blocked till this target is carried with the context, and more completely - when apiserversource follows the similar architecture as pingsource. 
Please advise, I'll be happy to go back to that previous commit. 
I'll also go through the pingsource model more thoroughly so that I can help with refactoring in the future.